### PR TITLE
Python: Azure AI Inference tracing SDK

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

Addresses: https://github.com/microsoft/semantic-kernel/issues/9413

### Description

The latest Azure AI Inference SDK has been released with the tracing package. We have decided to upgrade to the latest so that we will no longer need to instrument the Azure AI Inference connector with our own model diagnostics module.

### Contribution Checklist
1. Upgrade to the latest Azure AI Inference SDK with the tracing package.
2. Refactor the AI Inference connector to reduce duplicated code.
3. Some other minor fixes.

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 2867 | 4 :zzz: | 0 :x: | 0 :fire: | 1m 8s :stopwatch: |

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_base.py [hunk 1]
```diff
@@ -3,23 +3,111 @@
...
+        }
+
+        return class_mapping[client_type]
 
 
 @experimental_class
 class AzureAIInferenceBase(KernelBaseModel, ABC):
     """Azure AI Inference Chat Completion Service."""
 
-    MODEL_PROVIDER_NAME: ClassVar[str] = "azureai"
```
Comment (TaoChenOSU) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_base.py:18]: This is potentially breaking however this was not supposed to be use by customers.

File: python/samples/demos/telemetry/main.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ async def main(scenario: Literal["ai_service", "kernel_function", "auto_function
     with tracer.start_as_current_span("main") as current_span:
         print(f"Trace ID: {format_trace_id(current_span.get_span_context().trace_id)}")
 
-        stream = True
+        stream = False
```
Comment (moonbox3) [inline python/samples/demos/telemetry/main.py:143]: Is this change needed?

File: python/samples/demos/telemetry/scenarios.py [hunk 3]
```diff
@@ -22,6 +22,12 @@ def set_up_kernel() -> Kernel:
     # All built-in AI services are instrumented with telemetry.
     # Select any AI service to see the telemetry in action.
     kernel.add_service(OpenAIChatCompletion(service_id="open_ai"))
+    # kernel.add_service(
```
Comment (moonbox3) [inline python/samples/demos/telemetry/scenarios.py:25]: Is this code needed?

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_base.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_base.py:18]: The class has an `experimental` tag, so we should be okay.

File: python/samples/demos/telemetry/main.py (hunk 2 above)
Comment (TaoChenOSU) [reply python/samples/demos/telemetry/main.py:143]: Yes, because Azure AI Inference doesn't support streaming function calling yet.

File: python/samples/demos/telemetry/scenarios.py (hunk 3 above)
Comment (TaoChenOSU) [reply python/samples/demos/telemetry/scenarios.py:25]: This is for customers to know that they can easily switch between services to run this sample. I also have Google commented out below.

Comment (moonbox3) [reply python/samples/demos/telemetry/scenarios.py:25]: Should we add some comment like "The following code shows how to add other AI services to the kernel" ? It just looks like commented out code otherwise.

Comment (moonbox3) [reply python/samples/demos/telemetry/scenarios.py:25]: I think we should get in the habit (myself included) of overcommitting the code -- comments help people from all different backgrounds and skillsets.

**Review by moonbox3 — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]
